wasm-chain-integration/src/v1/trie/low_level.rs

```diff
@@ -1472,14 +1493,19 @@ impl<V> MutableTrie<V> {
         } else {
             return Ok(false);
         };
+        let generation = owned_nodes[node_idx].generation;
+        let iterator_roots = &mut self.iterator_roots;
+        if is_subtree_locked(
+            SubtreeAction::InsertOrDelete,
+            &iterator_roots[generation as usize],
+            key,
+        ) {
+            return Err(AttemptToModifyLockedArea);
+        }
```

Comment:
I recommend to rephrase this function in `check_subtree_locked` and have it return `Result<(), AttemptToModifyLockedArea>`. Then use `?`.